core/driver/sources/src/test/scala/org/virtuslab/ideprobe/dependencies/IntelliJCleanupTest.scala

```diff
@@ -0,0 +1,76 @@
+package org.virtuslab.ideprobe.dependencies
+
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.virtuslab.ideprobe.config.{DependenciesConfig, IdeProbeConfig, IntelliJProvider, IntellijConfig, PathsConfig}
+import org.virtuslab.ideprobe.ide.intellij.IntelliJFactory
+
+import java.nio.file.Path
+
+@RunWith(classOf[JUnit4])
+final class IntelliJCleanupTest {
+  @Test
+  def existingIntelliJShouldNotBeDeletedDuringCleanup: Unit = givenInstalledIntelliJ { installationRoot =>
+      val intelliJProvider = IntelliJProvider.from(
```

Comment:
it would be best to either create provider in one of two ways:
- instantiate fixture from config in form of string
- create IntelliJProvider using constructor i.e. programatically

Basically there are 2 ways to use probe, either from code using public api and helper `withX` methods or create it from config file/string.

Classes from config package are just implementation detail for parsing the config.

You can notice here also that IdeProbeConfig is too much to pass, as you specify `workspace` here that is unused in the IJProvider

Reply:
I'll do it in the next commit, together with the Factory renaming.

Reply:
Created a fixture (and therefore the provider) from config string.